tests/foreman/ui/test_contentview.py

```diff
@@ -877,6 +888,100 @@ def test_positive_update_exclusive_filter_package_version(self):
                 )
             )
 
+    @skip_if_bug_open('bugzilla', 1489778)
+    @tier2
+    def test_positive_errata_inc_update_list_package(self):
+        """Publish incremental update with a new errata for a custom repo
+
+        :BZ: 1489778
+
+        :id: fb43791c-60ee-4190-86be-34ccba411396
+
+        :expectedresults: New errata and corresponding package are present
+            in new content view version
+
+        :CaseImportance: High
+
+        :CaseLevel: Integration
+        """
+        # Create and publish a repo with 1 outdated package and some errata
+        repo_name = gen_string('alphanumeric')
+        repo_url = create_repo(
+            repo_name,
+            FAKE_0_INC_UPD,
+            [FAKE_0_INC_UPD_PACKAGES[0]]
+        )
+        result = repo_add_updateinfo(
+            repo_name, '{}{}'
+            .format(FAKE_0_INC_UPD, FAKE_0_INC_UPD_UPDATEFILES[0])
+        )
+        self.assertEqual(result.return_code, 0)
+        # Create org, product, repo, sync & publish it
+        org = make_org()
+        product = make_product({'organization-id': org['id']})
+        repo = make_repository({
+            'product-id': product['id'],
+            'url': repo_url,
+        })
+        Repository.synchronize({'id': repo['id']})
+        content_view = make_content_view({
+            'organization-id': org['id'],
+            'repository-ids': repo['id'],
+        })
+        ContentView.publish({'id': content_view['id']})
+        content_view = ContentView.info({'id': content_view['id']})
+        self.assertEqual(len(content_view['versions']), 1)
+        cvv = content_view['versions'][0]
+        # Add updated package to the repo and errata for the outdated package
+        create_repo(
```

Comment:
so you remove all previous data in repository and upload a new one? Why do we need the old data in that case?

Reply:
Not all the data but all repodata (contains of repodata/ folder) - all the packages inside the repo remain untouched. That's part of process of updating the repo, basically we take existing repo with all its packages, add extra packages/errata/whatever, remove metadata and generate it once more to include new packages.
Personally, i was a bit surprised too - if createrepo command updates repodata, then what's the point of wiping it (other than 'just in case')? But when i tried it out, scenario was not working, so i think there's a reason behind this, like remains of old repodata may conflict or something 🙂
Btw, that's what deploying custom errata scenario suggests, if you're interested, pls take a look at https://access.redhat.com/articles/1526413